src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -149,12 +148,37 @@ def right_size(
 
         self._init_sagemaker_session_if_does_not_exist()
 
+        self.temp_model_name = None
+        if isinstance(self, sagemaker.model.Model) and not isinstance(
+            self, sagemaker.model.ModelPackage
+        ):
+
+            unique_tail = uuid.uuid4()
+            self.temp_model_name = "SMPYTHONSDK-" + str(unique_tail)
```

Comment:
`"SMPYTHONSDK-" + str(unique_tail)` here has the same head: as https://github.com/aws/sagemaker-python-sdk/blob/6b4e6757f7c47ae63faa9910d7928672da4e9e22/src/sagemaker/session.py#L4989 can we distinguish the two?

Reply:
model name and job name will show up in the inference recommendations response right?